nlp/2021_07_07_neural_keyword_extraction/neural_keyword_extraction.ipynb

```diff
@@ -0,0 +1,1520 @@
+              "</table>\n",
+              "</div>"
+            ],
+            "text/plain": [
+              "              KeyBERT   BART-based  ...              TextRank       TopicRank\n",
+              "0            hot goat   Restaurant  ...  concentrated cooking   moreno family\n",
+              "1  deliciously sticky  Los Angeles  ...             goat meat    style birria\n",
+              "2           goat meat               ...              big bowl       zacatecan\n",
+              "3  places deliciously               ...          style birria          places\n",
+              "4   nochistlán moreno               ...  birrieria nochistlán          sticky\n",
+              "5            meat isn               ...                   los        big bowl\n",
+              "6        rosio moreno               ...                 salsa           thick\n",
+              "7       vibrant salsa               ...                moreno   hot goat meat\n",
+              "8           birrieria               ...                  meat            home\n",
+              "9      birria exactly               ...                birria  soft tortillas\n",
+              "\n",
+              "[10 rows x 6 columns]"
+            ]
+          },
+          "metadata": {
+            "tags": []
+          }
+        },
+        {
+          "output_type": "stream",
+          "text": [
+            "WARNING:root:LoadFile._df_counts is hard coded to /usr/local/lib/python3.7/dist-packages/pke/models/df-semeval2010.tsv.gz\n",
+            "WARNING:root:Candidates are generated using 0.33-top\n"
+          ],
+          "name": "stderr"
+        },
+        {
+          "output_type": "display_data",
+          "data": {
+            "text/html": [
+              "<div>\n",
+              "<style scoped>\n",
+              "    .dataframe tbody tr th:only-of-type {\n",
+              "        vertical-align: middle;\n",
+              "    }\n",
+              "\n",
+              "    .dataframe tbody tr th {\n",
+              "        vertical-align: top;\n",
+              "    }\n",
+              "\n",
+              "    .dataframe thead th {\n",
+              "        text-align: right;\n",
+              "    }\n",
+              "</style>\n",
+              "<table border=\"1\" class=\"dataframe\">\n",
+              "  <thead>\n",
+              "    <tr style=\"text-align: right;\">\n",
+              "      <th></th>\n",
+              "      <th>KeyBERT</th>\n",
+              "      <th>BART-based</th>\n",
+              "      <th>TFIDF</th>\n",
+              "      <th>YAKE</th>\n",
+              "      <th>TextRank</th>\n",
+              "      <th>TopicRank</th>\n",
+              "    </tr>\n",
+              "  </thead>\n",
+              "  <tbody>\n",
+              "    <tr>\n",
+              "      <th>0</th>\n",
+              "      <td>methods unable</td>\n",
+              "      <td>Search Engines</td>\n",
```

Comment:
Funny how the BART-based method seems to dream up certain keywords, saw this popping up quite some times when playing around with it. Would definitely mention this somewhere in the conclusion or in the intro on the BART-based methods

Reply:
I think this this depends on the dataset actually. One dataset only contains three keywords, the other contains more. Good remark though, I'll try to clarify this above!

Reply:
Do you mean that some of the words don't actually appear in the text? This is one of the potential benefits of the neural approaches: keyword generation or abstractive extraction of you will.

Reply:
That's also definitely an interesting way of looking at it, it would seem that the knife cuts both-ways:
- a benefit is indeed that it can create abstractive keywords
- a disadvantage is that the abstractive keywords are sometimes not perfectly related to the text in question (like in this case)